### ISSUE TITLE
node2nix: fatal: ambiguous argument 'master': unknown revision or path not in the working tree.

### ISSUE THREAD

**Issue by ElessarWebb** [svanderburg/node2nix#15]
Node2nix fails to write packages.json with the following error messages:

```
Cloning git repository: git://github.com/mapbox/mapbox-gl-style-spec
Cloning into 'mapbox-gl-style-spec'...
remote: Counting objects: 6227, done.
remote: Compressing objects: 100% (13/13), done.
remote: Total 6227 (delta 6), reused 0 (delta 0), pack-reused 6214
Receiving objects: 100% (6227/6227), 1.68 MiB | 2.52 MiB/s, done.
Resolving deltas: 100% (3421/3421), done.
Checking connectivity... done.
Parsing the revision of commitish: master
fatal: ambiguous argument 'master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
Parsing the revision of commitish: origin/master
fatal: ambiguous argument 'origin/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
Cannot find the corresponding revision of: master
```

I think it results from this (nested) dependency: `mapbox-gl/package.json:60:    "mapbox-gl-style-spec": "github:mapbox/mapbox-gl-style-spec#83b1a3e5837d785af582efd5ed1a212f2df6a4ae",`

On the other hand, the similar dependency `mapbox-gl/package.json:59:    "mapbox-gl-shaders": "github:mapbox/mapbox-gl-shaders#de2ab007455aa2587c552694c68583f94c9f2747",` seems to go through, although the shell reports:

```
Cloning git repository: git://github.com/mapbox/mapbox-gl-shaders
Cloning into 'mapbox-gl-shaders'...
remote: Counting objects: 1618, done.
remote: Total 1618 (delta 0), reused 0 (delta 0), pack-reused 1618
Receiving objects: 100% (1618/1618), 227.49 KiB | 0 bytes/s, done.
Resolving deltas: 100% (1199/1199), done.
Checking connectivity... done.
Parsing the revision of commitish: master
Checking out revision: 7971dc1705f20f453c3ed42f240f53ca39e31778
Note: checking out '7971dc1705f20f453c3ed42f240f53ca39e31778'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>
```

Which is the wrong revision!

**Comment by svanderburg** [comment]
I have implemented a fix for this in revision: 7e0221b83af391ec178337d55b99cec863201d67

You can give it a try, by cloning this repo and installing the development version with:

```bash
$ nix-env -f release.nix -iA build.x86_64-linux
```

**Comment by svanderburg** [comment]
I have changed this, tested this, and I consider it to be solved now
